File: arena-jface/src/main/java/org/uqbar/lacar/ui/impl/jface/JFaceApplicationBuilder.java
package org.uqbar.lacar.ui.impl.jface;

import org.eclipse.core.databinding.observable.Realm;
import org.eclipse.jface.databinding.swt.SWTObservables;
import org.eclipse.swt.widgets.Display;

import org.uqbar.lacar.ui.impl.jface.windows.JFaceWindowBuilder;
import org.uqbar.lacar.ui.model.ApplicationRunner;
import org.uqbar.lacar.ui.model.WindowBuilder;

/**
 * Implementacion de {@link ApplicationRunner} para jface.
 * 
 * @author npasserini
 */
public class JFaceApplicationBuilder implements ApplicationRunner {
	private Display display;

	public JFaceApplicationBuilder() {
		this.display = new Display();
	}

	@Override
	public void run(Runnable application) {
		Realm.runWithDefault(SWTObservables.getRealm(this.display), application);
		this.display.dispose();
	}

	@Override
	public WindowBuilder createWindow() {
		return new JFaceWindowBuilder();
	}

}

File: arena-jface/src/main/scala/org/uqbar/arena/jface/app/JFaceApplicationBuilder.scala
package org.uqbar.arena.jface.app

import org.eclipse.core.databinding.observable.Realm
import org.eclipse.jface.databinding.swt.SWTObservables
import org.eclipse.swt.widgets.Display
import org.uqbar.lacar.ui.model.ApplicationRunner
import org.uqbar.lacar.ui.impl.jface.builder.windows.JFaceWindowBuilder

/**
 * Implementacion de {@link ApplicationRunner} para jface.
 * @author npasserini
 * @author jfernandes
 */
class JFaceApplicationBuilder extends ApplicationRunner {
  var display = new Display();

  override def run(application: Runnable) {
    Realm.runWithDefault(SWTObservables.getRealm(display), application)
    display.dispose
  }

  override def createWindow() = new JFaceWindowBuilder()

}